routes/POST-profile.js

```diff
@@ -1,48 +1,111 @@
+			return res.redirect("/profile?tm=No files were uploaded&ts=false")
 		}
-		file = req.files.image;
-		fileExtension = file.name.substring(file.name.lastIndexOf(".") + 1);
-		fileKey = uuid() + "." + fileExtension;
+		file = req.files.image
+		fileExtension = file.name.substring(file.name.lastIndexOf(".") + 1)
+		fileKey = uuid() + "." + fileExtension
 		var params = {
 			Bucket: process.env.BUCKET_NAME,
 			Key: fileKey,
-			Body: file.data
-		};
-		s3Client.upload(params, function(err, data) {
-			if (err) { console.log(err); return res.status(500).json({ error: "Internal Server Error 500" }); }
+			Body: file.data,
+		}
+		s3Client.upload(params, function (err, data) {
+			if (err) {
+				console.log(err)
+				return res
+					.status(500)
+					.json({ error: "Internal Server Error 500" })
+			}
 			// Update Database
-			db.query(`SELECT * FROM cloudclub.profiles WHERE userid='${sesh.userid}'`, (err1, result1) => {
-				if (err1) {console.log(err1); return res.status(500).json({ error: "Internal Server Error 500" });}
-				else if (result1.length === 0) {
-					db.query("INSERT INTO cloudclub.profiles (`profile_path`, `description`, userid) VALUES (?, ?, ?)", [data.Location, "", sesh.userid], (err2) => {
-						if (err2) { console.log(err2); return res.status(500).json({ error: "Internal Server Error 500" }); }
-						else return res.status(201).redirect("/profile");
-					});
-				} 
-				else {
-					var oldProfile = result1[0]["profile_path"];
-					oldProfile = oldProfile.substring(oldProfile.lastIndexOf("/") + 1);
-					s3Client.deleteObject({Bucket: process.env.BUCKET_NAME, Key: oldProfile}, function(err) {
-						if (err) { console.log(err.stack); return res.status(500).json({ error: "Internal Server Error 500" }); }
-						db.query(`UPDATE cloudclub.profiles SET profile_path = '${data.Location}' WHERE userid='${sesh.userid}'`, (err2) => {
-							if (err2) { console.log(err2); return res.status(500).json({ error: "Internal Server Error 500" }); }
-							else return res.status(201).redirect("/profile");
-						});
-					});
+			db.query(
+				`SELECT * FROM cloudclub.profiles WHERE userid='${sesh.userid}'`,
+				(err1, result1) => {
+					if (err1) {
+						console.log(err1)
+						return res
+							.status(500)
+							.json({ error: "Internal Server Error 500" })
+					} else if (result1.length === 0) {
+						db.query(
+							"INSERT INTO cloudclub.profiles (`profile_path`, `description`, userid) VALUES (?, ?, ?)",
+							[data.Location, "", sesh.userid],
+							(err2) => {
+								if (err2) {
+									console.log(err2)
+									return res.status(500).json({
+										error: "Internal Server Error 500",
+									})
+								} else
+									return res.status(201).redirect("/profile")
+							}
+						)
+					} else {
+						var oldProfile = result1[0]["profile_path"]
+						oldProfile = oldProfile.substring(
+							oldProfile.lastIndexOf("/") + 1
+						)
+
+						if (oldProfile === "" || oldProfile === undefined) {
```

Comment:
Hi @Lepabryce ,

I think I may have figured out the bug. On line 59, oldProfile may have been empty or undefined for some users. This was giving an error where the Params.Key was empty (it was an empty string => '' ).